compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -326,7 +326,7 @@ impl<Prov: Provenance, Bytes: AllocBytes> Allocation<Prov, (), Bytes> {
     }
 
     /// Try to create an Allocation of `size` bytes, panics if there is not enough memory
-    /// available to the compiler to do so.
+    /// available to the compiler to do so. Also used for creating dummy `Allocation`.
```

Comment:
What's a dummy allocation?

The comment change is confusing me.

Reply:
Maybe I should documented as 
>can be used for creating allocations out of `Scalar` values, 

because from the current doc I couldn't understand I could do that I had to ask oli about it. It's very hard to understand what functions do or when to use them when you are foreign some part of the compiler.

Reply:
I think what you are asking for is a complete tutorial on the data model of the CTFE interpreter and its memory. That's reasonable, this should ideally be documented somewhere, but `fn uninit` is not the right place for that.

There's nothing specific about `uninit` that makes it more or less suited for "dummy" allocations so I think this comment change should be removed. Also note that "converting a Scalar to an Allocation" is an extremely uncommon usecase, so it might not be a good idea to gear the documentation towards that.

You could say something general like, "To obtain an allocation filled with specific data, first call this function and then call `write_scalar` to fill in the right data."

Reply:
Now on second thought, you are right maybe someone could explain these topics in depth on rustc dev guide